packages/app/obojobo-express/server/routes/api/locks.js

```diff
@@ -0,0 +1,69 @@
+const express = require('express')
+const router = express.Router()
+const EditLock = oboRequire('server/models/edit_lock')
+const { userHasPermissionToDraft } = require('obojobo-repository/server/services/permissions')
+const { checkValidationRules, requireDraftId, requireContentId, requireCanViewEditor } = oboRequire(
+	'server/express_validators'
+)
+
+// CHECK A LOCK
+// mounted as /api/locks/:draftId
+router
+	.route('/:draftId')
+	.get([requireDraftId, requireCanViewEditor, checkValidationRules])
+	.get(async (req, res) => {
+		const existingLock = await EditLock.fetchByDraftId(req.params.draftId)
+		res.success(existingLock)
+	})
+
+// CREATE A LOCK
+// mounted as /api/locks/:draftId
+router
+	.route('/:draftId')
+	.post([requireDraftId, requireCanViewEditor, requireContentId, checkValidationRules])
+	.post(async (req, res) => {
+		try {
+			const hasPerms = await userHasPermissionToDraft(req.currentUser.id, req.params.draftId)
+
+			if (!hasPerms) {
+				return res.notAuthorized('Authoring permissions required to create a lock.')
+			}
+
+			const existingLock = await EditLock.fetchByDraftId(req.params.draftId)
+
+			if (existingLock && existingLock.userId !== req.currentUser.id) {
+				return res.notAuthorized('Draft is already locked by someone else.')
+			}
+
+			const myLock = await EditLock.create(
+				req.currentUser.id,
+				req.params.draftId,
+				req.body.contentId
+			)
+
```

Comment:
What about this - instead of 

```javascript
existingLock = EditLock.fetchByDraftId(...)
if(existingLock and existingLock.userId is not me) throw 'Already locked'
myLock = EditLock.create(...)
```

We do

```javascript
myLock = EditLock.create()
if(myLock is empty or false or the userId is not my userId) throw 'Already locked'
```

Since EditLock.create has code to prevent anything from being inserted if a lock already exists (unless I'm mis-understanding the query)